### PR TITLE
support SV3D: the multiview video generation with 3D coherence

### PR DESCRIPTION
**Pull request by HaFred**
# What does this PR do?
This is the mindspore implementation for [SV3D](https://huggingface.co/stabilityai/sv3d).
| Image 3  | 
| :-----------------: |
| Output Multiview Images (21x576x576) |

## Before submitting
- [x] Did you read the [contributor guideline](https://github.com/mindspore-lab/mindone/blob/master/CONTRIBUTING.md)?
- [x] Did you make sure to update the documentation with your changes? E.g. record bug fixes or new features in `What's New`. Here are the
      [documentation guidelines](https://github.com/mindspore-lab/mindcv/wiki/%E6%96%87%E6%A1%A3%E7%BC%96%E5%86%99%E8%A7%84%E8%8C%83)
- [x] Did you build and run the code without any errors?
- [x] Did you report the running environment (NPU type/MS version) and performance in the doc? (better record it for data loading, model inference, or training tasks)
- [x] Did you write any new necessary tests?

## Who can review?

Plz kindly review and comment, thank you Jason. @SamitHuang 

Fred

### CODE REVIEW

**Comment by SamitHuang** [comment]
nice job. here are some suggestions.
1. remove redundant files to keep clean code & docs.  data/test_image.png, sgm/dataset_cache.py, etc
2. supplement the REAME with the verified environments (chip, ms version, requirements, etc) and the generated 3D samples.
3. run pre-commit to fix the format issues

**Comment by HaFred** [comment]
ok to merge now, thx

File: examples/sv3d/simple_video_sample.py [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+from utils import mixed_precision
+
+import mindspore as ms
+from mindspore import Tensor, nn, ops
+from mindspore.dataset.vision import ToTensor
+
+from mindone.utils.seed import set_random_seed
+
+
+class SVD3InferPipeline(nn.Cell):
```
Comment (hadipash) [inline examples/sv3d/simple_video_sample.py:47]: `SV3DInferPipeline`?

Comment (HaFred) [reply examples/sv3d/simple_video_sample.py:47]: > `SV3DInferPipeline`?

Hi Rustam thank you so much for the reviewing efforts. The typo has been fixed thanks.

File: examples/sv3d/README.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+Vendor ID:                       HiSilicon
+Model name:                      Kunpeng-920
+NPU:                             910B1
+Mindspore Version:               2.3.0.B528
+CANN Version:                    7.3
+Ascend Driver:                   23.0.rc3.6
+```
+
+## Pretrained Models
+- [ ] Mindspore Checkpoint Release
```
Comment (SamitHuang) [inline examples/sv3d/README.md:57]: what about providing the checkpoint conversion script?

File: examples/sv3d/configs/sampling/sv3d_u.yaml [hunk 3]
```diff
@@ -0,0 +1,124 @@
+model:
+  target: models.diffusion.VideoDiffusionEngine
```
Comment (SamitHuang) [inline examples/sv3d/configs/sampling/sv3d_u.yaml:2]: The name VideoDiffusionEngine seems confusing. Why not MultiViewDiffusionEngine?

File: examples/sv3d/configs/sampling/sv3d_u.yaml [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+      params:
+        weighting_config:
+          target: sgm.modules.diffusionmodules.denoiser_weighting.EDMWeighting
+          params:
+            sigma_data: 1.0
+        scaling_config:
+          target: sgm.modules.diffusionmodules.denoiser_scaling.VScalingWithEDMcNoise
+
+    network_config:
+      target: modules.video_model.VideoUNet
```
Comment (SamitHuang) [inline examples/sv3d/configs/sampling/sv3d_u.yaml:19]: same here. video_*

File: examples/sv3d/configs/sampling/sv3d_u.yaml [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+            n_cond_frames: 1
+            n_copies: 1
+            open_clip_embedding_config:
+              target: sgm.modules.embedders.modules.FrozenOpenCLIPImageEmbedder
+              params:
+                freeze: True
+
+        - input_key: cond_frames
+          is_trainable: False
+          target: sgm.modules.embedders.modules.VideoPredictionEmbedderWithEncoder
```
Comment (SamitHuang) [inline examples/sv3d/configs/sampling/sv3d_u.yaml:56]: same here.

File: examples/sv3d/requirements.txt [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+onnxruntime==1.18.0
+op-compile-tool==0.1.0
+opc-tool==0.1.0
+opencv-python==4.10.0.82
+opencv-python-headless==4.10.0.82
+packaging==24.1
+pandas==2.2.2
+pathspec==0.12.1
+pillow==10.2.0
+pip==24.0
```
Comment (SamitHuang) [inline examples/sv3d/requirements.txt:75]: pls remove redundant packages

File: examples/sv3d/sgm/data/dataset_wds.py
Comment (SamitHuang) [inline examples/sv3d/sgm/data/dataset_wds.py:1]: is it used?

File: examples/sv3d/sgm/data/dataset_cache.py
Comment (SamitHuang) [inline examples/sv3d/sgm/data/dataset_cache.py:1]: is it used?

**Review by SamitHuang — COMMENTED** [review]
The documents are improved well. To make it follow open-source standard:
1. There are a lot of files and classes name with Video* prefix, which is inappropriate. Pls re-organize.
2. Redundant files that are not used should be removed if they are tested in this 3D generation tasks, for reliability and clean code.

File: examples/sv3d/configs/sampling/sv3d_u.yaml (hunk 3 above)
Comment (SamitHuang) [reply examples/sv3d/configs/sampling/sv3d_u.yaml:2]: or simply DiffusionEngine

File: examples/sv3d/requirements.txt (hunk 6 above)
Comment (HaFred) [reply examples/sv3d/requirements.txt:75]: No problem.

File: examples/sv3d/README.md (hunk 2 above)
Comment (HaFred) [reply examples/sv3d/README.md:57]: No problem, I will update the readme accordingly.

File: examples/sv3d/configs/sampling/sv3d_u.yaml (hunk 3 above)
Comment (HaFred) [reply examples/sv3d/configs/sampling/sv3d_u.yaml:2]: Agree, I will put it as `MultiviewVideoDiffusionEngine` for better readability.

File: examples/sv3d/configs/sampling/sv3d_u.yaml (hunk 4 above)
Comment (HaFred) [reply examples/sv3d/configs/sampling/sv3d_u.yaml:19]: I am neutral about whether to change it, as I tried to follow the official naming here: https://github.com/Stability-AI/generative-models/blob/main/configs/inference/sv3d_u.yaml#L14.

File: examples/sv3d/configs/sampling/sv3d_u.yaml (hunk 5 above)
Comment (HaFred) [reply examples/sv3d/configs/sampling/sv3d_u.yaml:56]: Same above.

File: examples/sv3d/sgm/data/dataset_cache.py
Comment (HaFred) [reply examples/sv3d/sgm/data/dataset_cache.py:1]: deprecated. will remove

**Pull request closed without merging**